src/renderer/components/SelectAccount.tsx

```diff
@@ -0,0 +1,368 @@
+  const unit = getAccountUnit(account);
+  const name = getAccountName(account);
+  const nested = ["TokenAccount", "ChildAccount"].includes(account.type);
+  const balance =
+    account.type !== "ChildAccount" && account.spendableBalance
+      ? account.spendableBalance
+      : account.balance;
+
+  return (
+    <Flex flexGrow={1} alignItems="center" columnGap={4} height={48}>
+      {!isSelectedValue && nested ? <Tick /> : null}
+      {!isSelectedValue && (
+        <Text color="neutral.c100">
+          <CryptoCurrencyIcon currency={currency} size={16} />
+        </Text>
+      )}
+      <Text flex={1} ff="Inter|SemiBold" color="inherit" fontSize={4}>
+        {name}
+      </Text>
+      {!hideDerivationTag && <AccountTagDerivationMode account={account} />}
+      {!isSelectedValue && unit && balance && (
+        <Text color="neutral.c100" mr={4}>
+          <FormattedVal
+            color="palette.text.shade40"
+            ff="Inter|Medium"
+            fontSize={3}
+            val={balance}
+            unit={unit}
+            showCode
+          />
+        </Text>
+      )}
+    </Flex>
+  );
+};
+
+const defaultRenderOption = (props: OptionProps<SelectOption, false>) => {
+  return (
+    <SelectOption {...props} render={({ data }) => <AccountOption account={data.account} />} />
+  );
+};
+
+const defaultRenderValue = (props: SingleValueProps<SelectOption>) => {
+  return (
+    <components.SingleValue {...props}>
+      <AccountOption account={props.getValue()[0].account} isSelectedValue />
+    </components.SingleValue>
+  );
+};
+
+const AddAccountContainer = styled(Box)`
+  position: relative;
+  cursor: pointer;
+  flex-direction: row;
+  align-items: center;
+  border-top: 1px solid ${p => p.theme.colors.neutral.c40};
+  border-radius: 0 0 8px 8px;
+  overflow: hidden;
+  background: ${p => p.theme.colors.background.main};
+`;
+
+const AddButton = styled(Button.Unstyled)`
+  width: 100%;
+  height: 48px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  &:hover {
+    background-color: ${p => p.theme.colors.primary.c20};
+  }
+  &:active {
+    background-color: ${p => p.theme.colors.primary.c30};
+  }
+`;
+
+const AddAccountFooter = (props: any) => {
+  const dispatch = useDispatch();
+  const openAddAccounts = useCallback(() => {
+    dispatch(openModal("MODAL_ADD_ACCOUNTS", null));
+  }, [dispatch]);
+
+  return (
+    <>
+      <components.MenuList {...props} />
+      <AddAccountContainer>
+        <AddButton onClick={openAddAccounts}>
+          <Text color="neutral.c80">
+            <Plus size={12} />
+            <span style={{ marginLeft: 8 }}>
+              <Trans i18nKey="swap2.form.details.noAccountCTA" />
+            </span>
+          </Text>
+        </AddButton>
+      </AddAccountContainer>
+    </>
+  );
+};
+
+const renderLeft = (props: SelectInputProps<SelectOption>) => {
+  const account: AccountLike = props.getValue()[0].account;
```

Comment:
⚠️ `props.getValue()[0]` could be null I think

Reply:
Done ✅ I made proper checks and wrapped into `getAccountFromProps` function